File: server/node/test/hz/message/checker_test.js
#!/usr/bin/env node

'use strict';

const assert = require('assert');
const checker = require('../../../src/hz/message/checker.js');

describe('checker.checkParamType()', function() {
	it('测试空参数', function() {
		let [val, err] = checker.checkParamType({});
		assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法key参数', function() {
		let [val, err] = checker.checkParamType({abcd:'1234'});
        assert(val == undefined && err.status == 'lost_parameter');
		[val, err] = checker.checkParamType({Type:'1234'});
        assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法value参数', function() {
		let [val, err] = checker.checkParamType({type:1234});
        assert(val == undefined && err.status == 'invalid_parameter');
		[val, err] = checker.checkParamType({type:''});
        assert(val == undefined && err.status == 'invalid_parameter');
	});
	
	it('测试合法参数', function() {
		let [val, err] = checker.checkParamType({type:'1234'});
        assert(val == '1234' && err == undefined);
	});
});

describe('checker.checkParamId()', function() {
	it('测试空参数', function() {
		let [val, err] = checker.checkParamId({});
		assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法key参数', function() {
		let [val, err] = checker.checkParamId({abcd:1234});
		assert(val == undefined && err.status == 'lost_parameter');
		[val, err] = checker.checkParamId({Id:1234});
		assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法value参数', function() {
		let [val, err] = checker.checkParamId({id:'1234'});
		assert(val == undefined && err.status == 'invalid_parameter');
	});
	
	it('测试合法参数', function() {
		let [val, err] = checker.checkParamId({id:1234});
		assert(val == 1234 && err == undefined);
	});
});

describe('checker.checkParamContent()', function() {
	it('测试空参数', function() {
		let [val, err] = checker.checkParamContent({});
		assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法key参数', function() {
		let [val, err] = checker.checkParamContent({abcd:'1234'});
		assert(val == undefined && err.status == 'lost_parameter');
		[val, err] = checker.checkParamContent({Content:'1234'});
		assert(val == undefined && err.status == 'lost_parameter');
	});
	
	it('测试非法value参数', function() {
		let [val, err] = checker.checkParamContent({content:1234});
		assert(val == undefined && err.status == 'invalid_parameter');
		[val, err] = checker.checkParamContent({content:''});
		assert(val == undefined && err.status == 'invalid_parameter');
	});
	
	it('测试合法参数', function() {
		let [val, err] = checker.checkParamContent({content:'1234'});
		assert(val == '1234' && err == undefined);
	});
});

File: server/java/test/hz/message/CheckerTest.java
package hz.message;

import static org.junit.Assert.*;

import org.junit.Test;
import org.json.JSONObject;

/**
 * 测试Checker
 */
public class CheckerTest {
	/**
	 * 测试Checker.checkParamType()
	 */
	@Test
	public void test_checkParamType() {
		// 测试空参数
		Object[] res = Checker.checkParamType(new JSONObject("{}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		
		// 测试非法key参数
		res = Checker.checkParamType(new JSONObject("{\"abcd\": \"1234\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		res = Checker.checkParamType(new JSONObject("{\"Type\": \"1234\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		
		// 测试非法value参数
		res = Checker.checkParamType(new JSONObject("{\"type\": 1234}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("invalid_parameter"));
		res = Checker.checkParamType(new JSONObject("{\"type\": \"\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("invalid_parameter"));
		
		// 测试合法参数
		res = Checker.checkParamType(new JSONObject("{\"type\": \"1234\"}"));
		assertTrue(res[0].equals("1234") && res[1] == null);
	}
	
	/**
	 * 测试Checker.checkParamId()
	 */
	@Test
	public void test_checkParamId() {
		// 测试空参数
		Object[] res = Checker.checkParamId(new JSONObject("{}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		
		// 测试非法key参数
		res = Checker.checkParamId(new JSONObject("{\"abcd\": 1234}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		res = Checker.checkParamId(new JSONObject("{\"Id\": 1234}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));

		// 测试非法value参数
		res = Checker.checkParamId(new JSONObject("{\"id\": \"1234\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("invalid_parameter"));
		
		// 测试合法参数
		res = Checker.checkParamId(new JSONObject("{\"id\": 1234}"));
		assertTrue(res[0].equals(1234) && res[1] == null);
	}
	
	/**
	 * 测试Checker.checkParamContent()
	 */
	@Test
	public void test_checkParamContent() {
		// 测试空参数
		Object[] res = Checker.checkParamContent(new JSONObject("{}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));

		// 测试非法key参数
		res = Checker.checkParamContent(new JSONObject("{\"abcd\": \"1234\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));
		res = Checker.checkParamContent(new JSONObject("{\"Content\": \"1234\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("lost_parameter"));

		// 测试非法value参数
		res = Checker.checkParamContent(new JSONObject("{\"content\": 1234}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("invalid_parameter"));
		res = Checker.checkParamContent(new JSONObject("{\"content\": \"\"}"));
		assertTrue(res[0] == null && ((JSONObject) res[1]).getString("status").equals("invalid_parameter"));

		// 测试合法参数
		res = Checker.checkParamContent(new JSONObject("{\"content\": \"1234\"}"));
		assertTrue(res[0].equals("1234") && res[1] == null);
	}
}
